dg/tools-frameworks.md

```diff
@@ -0,0 +1,139 @@
+     throwIfFalse(hasOpenPhases,
+       () => new Error('There are no accessible phases.')));
+ }
+```
+
+For more information on the RxJS library, visit the links below.
+
+### Resources for Angular
+
+1. [Angular Guide](https://angular.io/guide/architecture) : Official Angular developer guide and introduction to basic Angular topics
+2. [RxJS Guide](https://rxjs-dev.firebaseapp.com/guide/observable) : Official RxJS guide on Observables, Observers, Operators, Subscription, etc.
+3. [Angular Guide on Navigation of Component Tree](https://angular.io/guide/dependency-injection-navtree) : Guide on how to navigate the component tree with Dependency Injection
+4. [Angular Tutorial](https://angular.io/tutorial) : Official Angular tutorials
+
+## Introduction to Electron
+
+Electron is a tool that allows developers to build cross-platform desktop applications with Javascript, HTML, and CSS. It basically allows us to "convert" the web version of CATcher into a desktop application through the APIs provided by Electron.
+
+### How Electron Works
+Electron uses the module `ipcMain` to communicate to renderer processes from the main process. It sends out and listen for events, enabling inter-process communication within the CATcher.
+
+Code snippet of `ipcMain` in CATcher
+```
+ipcMain.on('github-oauth', (event, repoPermissionLevel) => {
+  Logger.info('Starting authentication');
+  getAccessToken(win, repoPermissionLevel).then((data) => {
+    Logger.info('Obtained access token from Github');
+    event.sender.send('github-oauth-reply', {token: data.token});
+  }).catch(error => {
+    event.sender.send('github-oauth-reply', {
+      error: error.message,
+      isWindowClosed: error.message === 'WINDOW_CLOSED'});
+  });
+});
+```
+This starts the authentication flow in the CATcher desktop version. `ipcMain` communicates with the renderer processes by waiting for events from them, as well as emitting events to them.
+
+Because it is a desktop application, it is important that we account for different operating systems in our code, since different operating systems would behave differently to changes in our application.
+
+Example of adapting application logic to Linux O.S.
+```
+  if (isLinuxOs()) {
+    // app icon needs to be set manually on Linux platforms
+    windowOptions['icon'] = ICON_PATH;
+  }
+```
+
+### Resources for Electron
+
+1. [Official Electron Guide](https://www.electronjs.org/docs/tutorial) : This is the official Electron documentation]
+
+
+## Introduction to GraphQL
+
+GraphQL is a query language for APIs.
+
+In CATcher, it is to communicate with the Github backend. It allows us to define a schema for the data that we receive from github, and also performs validation for the data we reqeust for, such that the data we receive from Github is exactly what we need, nothing more, nothing less. The extra layer of GraphQL is one that makes the process of data retrieval in CATcher more streamlined.
```

Comment:
Can mention that the GraphQL queries used by CATcher are defined under the `graphql` folder in the codebase.